app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -229,48 +229,39 @@ def save_host_storages_inventory(host, hashes, target = nil)
     end
 
     host.host_storages(true)
-    deletes =
-      if target == host
-        host.host_storages.dup
-      else
-        []
-      end
+    deletes = if disconnect && target == host
+                host.host_storages.dup
+              else
+                []
+              end
 
     save_inventory_multi(host.host_storages, hashes, deletes, [:host_id, :storage_id], nil, [:storage])
   end
 
-  def save_folders_inventory(ems, hashes, target = nil)
+  def save_folders_inventory(ems, hashes, target = nil, disconnect = true)
     target = ems if target.nil?
 
     ems.ems_folders.reset
-    deletes = if (target == ems)
-                :use_association
-              else
-                []
-              end
+    deletes = disconnect_using_association(ems, target, disconnect)
 
     save_inventory_multi(ems.ems_folders, hashes, deletes, [:uid_ems], nil, :ems_children)
     store_ids_for_new_records(ems.ems_folders, hashes, :uid_ems)
   end
   alias_method :save_ems_folders_inventory, :save_folders_inventory
 
-  def save_clusters_inventory(ems, hashes, target = nil)
+  def save_clusters_inventory(ems, hashes, target = nil, disconnect = true)
     target = ems if target.nil?
 
     ems.ems_clusters.reset
-    deletes = if (target == ems)
-                :use_association
-              else
-                []
-              end
+    deletes = disconnect_using_association(ems, target, disconnect)
 
     save_inventory_multi(ems.ems_clusters, hashes, deletes, [:uid_ems], nil, :ems_children)
     store_ids_for_new_records(ems.ems_clusters, hashes, :uid_ems)
   end
   alias_method :save_ems_clusters_inventory, :save_clusters_inventory
 
-  def save_resource_pools_inventory(ems, hashes, target = nil)
-    target = ems if target.nil?
+  def save_resource_pools_inventory(ems, hashes, target = nil, disconnect = true)
+    target = ems if target.nil? && disconnect
```

Comment:
Because the disconnect logic for resource pools is more complex than the normal `if target == ems then :use_association else []`
The target has to be nil if disconnect is true so that we don't hit [this case](https://github.com/agrare/manageiq/blob/40fbcf24e61f9581d848c458ebfccb163d7d41ec/app/models/ems_refresh/save_inventory_infra.rb#L269-L271).  I had to do the same for [hosts](https://github.com/ManageIQ/manageiq/pull/14247/files/40fbcf24e61f9581d848c458ebfccb163d7d41ec#diff-50f9feed046a3410ac5531492059a6fbR117) and [vms](https://github.com/ManageIQ/manageiq/pull/14247/files/40fbcf24e61f9581d848c458ebfccb163d7d41ec#diff-090b07f911cee85c2acff15e6bda1032R360) because their disconnect logic is special.